spec/models/gids_redis_spec.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'common/exceptions'
+
+describe GIDSRedis do
+  let(:scrubbed_params) { {} }
+  let(:service) { GIDSRedis.new }
+  let(:body) { {} }
+  let(:gi_response) do
+    GI::Responses::GIDSResponse.new(status: 200, body: body)
+  end
+
+  it 'method_missing' do
+    allow_any_instance_of(GI::Client).to receive(:get_institution_details).and_return(gi_response)
+
+    expect(service.get_institution_details(scrubbed_params)).to eq(gi_response.body)
+  end
+end
```

Comment:
Can we get some more spec?
Show that it caches responses.
What happens when an error occurs when reaching out to GIDS?

Reply:
Added caching test

Reply:
Service displays standard error response there is a GIDS outage otherwise return vets-api error message and stack trace of internal error.